src/client/network.rs

```diff
@@ -21,20 +21,26 @@ use error::Result;
 /// Security relevant configuration options for `KafkaClient`.
 // This will be expanded in the future. See #51.
 #[cfg(feature = "security")]
-pub struct SecurityConfig(SslConnector);
+pub struct SecurityConfig {
+    connector: SslConnector,
+    verify_server_name: bool,
+}
 
 #[cfg(feature = "security")]
 impl SecurityConfig {
     /// In the future this will also support a kerbos via #51.
-    pub fn new(connector: SslConnector) -> SecurityConfig {
-        SecurityConfig(connector)
+    pub fn new(connector: SslConnector, verify_server_name: bool) -> SecurityConfig {
+        SecurityConfig {
+            connector: connector,
+            verify_server_name: verify_server_name,
+        }
```

Comment:
i would prefer we keep the original one parameter `new` method (defaulting `verify_server_name` to `true`), but provide a additional mutator for `SecurityConfig`:

```rust
impl SecurityConfig {
pub fn new(connector: SslConnector) -> Self {
        SecurityConfig(connector, true)
}
pub fn with_hostname_verification(mut self, verify_hostname: bool) -> Self {
  self.verify_server_name = verify_hostname;
  self
}
```

1. typically people will want hostname verification turned on by default, and ...
2. if not, they won't mind having to write a little bit more code.
3. this will also provide some kind of compatibility.

Generally speaking, we do not want to advertise or make it particularly easy to lower the security, just to provide the possibility if people know what they're doing. I also wouldn't mind if you add some comment to the `with_...` method about "dark forces at work" when using `with_hostname_verification(false)` ;)